### PR TITLE
[Feature-New-Edged] add register and patch node upstream operation

### PR DESCRIPTION
**Pull request by Shelley-BaoYue**
Signed-off-by: Shelley-BaoYue <baoyue2@huawei.com>

**What type of PR is this?**
/kind feature

#4157 

**What this PR does / why we need it**:
Part of pr for [new edged].

Add create node and patch node upstream operation for lite-Kubelet. Create node method is used for register node in cluster, and Kubelet use patch method to update node status. 

In order not to affect the status update of old node,  the original updateNodeStatus method is also retained. Patch method is introduced here for new edged.

### CODE REVIEW

File: pkg/apis/componentconfig/cloudcore/v1alpha1/types.go [hunk 1]
```diff
@@ -254,6 +254,12 @@ type EdgeControllerBuffer struct {
 	// QueryVolumeAttachment indicates the buffer of query volume attachment
 	// default 1024
 	QueryVolumeAttachment int32 `json:"queryVolumeAttachment,omitempty"`
+	// CreateNode indicates the buffer of create node
+	// default 1024
+	CreateNode int32 `json:"updateNode,omitempty"`
```
[the remark this answers is not in the document]
Comment (Shelley-BaoYue) [reply pkg/apis/componentconfig/cloudcore/v1alpha1/types.go]: > here, json: "createNode"

fixed.

**Comment by wackxu** [comment]
> In order not to affect the status update of old node, the original updatePodStatus method is also retained. Patch method is introduced here for new edged.

updatePodStatus -> updateNodeStatus

File: staging/src/github.com/kubeedge/beehive/pkg/core/model/message.go [hunk 2]
```diff
@@ -28,6 +29,7 @@ const (
 	ResourceTypePodlist             = "podlist"
 	ResourceTypePodStatus           = "podstatus"
 	ResourceTypeNodeStatus          = "nodestatus"
+	ResourceTypeNodePatch           = "nodepatch"
```
Comment (wackxu) [inline staging/src/github.com/kubeedge/beehive/pkg/core/model/message.go:32]: this seems not a resource type, the resource is nodestatus and the operation is patch

**Comment by wackxu** [comment]
mark updateNodeStatus  deprecated add TODO for updateNodeStatus func since that this will be deleted in the subsequent versions

**Comment by Shelley-BaoYue** [comment]
> mark updateNodeStatus deprecated add TODO for updateNodeStatus func since that this will be deleted in the subsequent versions

Done.

**Comment by Shelley-BaoYue** [comment]
> > In order not to affect the status update of old node, the original updatePodStatus method is also retained. Patch method is introduced here for new edged.
> 
> updatePodStatus -> updateNodeStatus

Done.

File: edge/pkg/edged/kubeclientbridge/typed/core/v1/node_bridge.go [hunk 3]
```diff
@@ -1,33 +1,11 @@
...
-See the License for the specific language governing permissions and
-limitations under the License.
-
-@CHANGELOG
-KubeEdge Authors: To make a bridge between kubeclient and metaclient,
-This file is derived from K8S client-go code with reduced set of methods
-Changes done are
-1. Package v1 got some functions from "k8s.io/client-go/kubernetes/typed/core/v1/fake/fake_node.go"
-and made some variant
-*/
```
Comment (fisherxu) [inline edge/pkg/edged/kubeclientbridge/typed/core/v1/node_bridge.go:22]: Why remove the License header? :)

Comment (Shelley-BaoYue) [reply edge/pkg/edged/kubeclientbridge/typed/core/v1/node_bridge.go]: > Why remove the License header? :)

 It was deleted by mistake.  😄

**Review by fisherxu — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**